src/pages/ClubDetail/ClubDetail.tsx

```diff
@@ -15,57 +18,112 @@ export const ClubDetail: React.FC = () => {
 
   const [clubDetail, setClubDetail] = useState<ClubType | null>(null);
 
-  const handleDelete = async () => {
-    try {
-      const isDeletionConfirmed = window.confirm(
-        'Are you sure you want to delete this club?',
-      );
-
-      if (isDeletionConfirmed) {
-        await deleteClub(Number(id));
-
-        navigate('/joinclub');
-      }
-    } catch (error) {
-      console.error('Error deleting club:', error);
-    }
-  };
-
   useEffect(() => {
     const fetchClubDetail = async () => {
-      try {
-        const clubData = await getClubDetail(Number(id));
+      const clubData = await getClubDetail(Number(id));
 
-        setClubDetail(clubData);
-      } catch (error) {
-        console.error('Error getting club detail:', error);
-      }
+      setClubDetail(clubData);
     };
 
     fetchClubDetail();
   }, [id]);
 
+  const handleDelete = async () => {
+    const isDeletionConfirmed = window.confirm(
+      'Are you sure you want to delete this club?',
+    );
+
+    if (isDeletionConfirmed) {
+      await deleteClub(Number(id));
+
+      navigate('/joinclub');
+    }
+  };
+
+  const handleUpdate = async (updatedData: Partial<ClubType>) => {
+    await updateClub(Number(id), updatedData);
```

Comment:
I am sure, the updateClub can return updated entity, so you don't have to get it right after